File: Contests/CPC UNJFSC/FIISI Code Sep 2024/D. Número Curioso.cpp
#include <bits/stdc++.h>

using namespace std;

void solve() {
  int numero; cin >> numero; numero++;

  while (true) {
    string num = to_string(numero);
    set<char> digitos;
    bool esCurioso = true;

    for (int i=0; i<num.size(); i++) {
      if (digitos.count(num[i]) > 0) {
        esCurioso = false;
        break;
      }
      digitos.insert(num[i]);
    }

    if (esCurioso) {
      cout << num << endl;
      return;
    }

    numero++;
  }
}

int main() {
  int numCasos; cin >> numCasos;
  while (numCasos--) {
    solve();
  }

  return 0;
}

File: Contests/CPC UNJFSC/FIISI Code Sep 2024/D. Número Curioso.java
import java.util.*;

public class Main {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    // leer numero de casos
    int t = scanner.nextInt();
    scanner.nextLine();
    
    for (int i = 0; i < t; i++) {
      solve(scanner);
    }
    
    scanner.close();
  }

  private static void solve(Scanner scanner) {
    // leer el numero inicial
    int numero = scanner.nextInt();
    numero++; // incrementar para buscar el siguiente numero curioso
    
    while (true) {
      String num = Integer.toString(numero);
      Set<Character> digitos = new HashSet<>();
      boolean esCurioso = true;

      // Verificar si los digitos son distintos
      for (int i = 0; i < num.length(); i++) {
        if (digitos.contains(num.charAt(i))) {
          esCurioso = false;
          break;
        }
        digitos.add(num.charAt(i));
      }

      // Si es curioso, imprimir el numero y salir del bucle
      if (esCurioso) {
        System.out.println(num);
        return;
      }

      numero++; // incrementar el numero para continuar buscando
    }
  }
}
